### ISSUE TITLE
sender.source variable is UNDEFINED when source is Discord

### ISSUE THREAD

**Issue by chiefstank** [sogehige/sogeBot#3934]
2020-06-23T15:47:04.144 DEBUG: * ======= COPY DEBUG MESSAGE FROM HERE =======
2020-06-23T15:47:04.145 DEBUG: * GENERAL      | OS: npm/6.14.4 node/v12.16.2 win32 x64
2020-06-23T15:47:04.211 DEBUG: *              | Bot version: 11.8.1
2020-06-23T15:47:04.211 DEBUG: *              | DB: sqlite
2020-06-23T15:47:04.212 DEBUG: *              | HEAP: 77.06 MB
2020-06-23T15:47:04.212 DEBUG: *              | Uptime: 07:49:59
2020-06-23T15:47:04.213 DEBUG: *              | Language: en
2020-06-23T15:47:04.213 DEBUG: *              | Mute: false
2020-06-23T15:47:04.213 DEBUG: * SYSTEMS      | -Songs, -Highlights, -Points, -Price, -Ranks, CustomCommands, Alias, -Bets, -Checklist, -Commercial, -Cooldown, -HowLongToBeat, -Keywords, -Moderation, Polls, -Queue, -Quotes, Raffles, -Scrim, Timers, -Top, UserInfo
2020-06-23T15:47:04.214 DEBUG: * GAMES        | -Duel, -FightMe, -Gamble, -Heist, -Roulette, -Seppuku
2020-06-23T15:47:04.214 DEBUG: * INTEGRATIONS | -Spotify, Discord, -ResponsiveVoice, -Donationalerts, -PhillipsHue, -Qiwi, -StreamElements, Streamlabs, -TipeeeStream, -Twitter
2020-06-23T15:47:04.214 DEBUG: * WIDGETS      | chat, eventlist
2020-06-23T15:47:04.215 DEBUG: * OAUTH        | BOT true | BROADCASTER true
2020-06-23T15:47:04.215 DEBUG: * ======= END OF DEBUG MESSAGE =======

Did test of the new "sender.source" variable using a custom variable script. This is the test script for the custom variable.

```
if (sender.source !== "twitch")
	return "source: "+sender.source
```

If the source was DISCORD the sender.source variable is UNDEFINED instead of 'discord'. Not sure if it matters, but I'm triggering the command in discord as the caster

**Comment by sogehige** [comment]
I don't seem to be able to reproduce this issue

My setup:
Custom command: `!test -> $_test`
Custom variable: `$_test -> eval -> change permission to viewers`
```
return sender.source;
```

Returns correctly discord or twitch source

**Comment by chiefstank** [comment]
I discovered more when testing.

I created new variables/commands like the one you outlined to test and it worked fine so I looked back at what was different about mine. The only thing, which I failed to mention, was that my command (that's not working) has parameters. If a parameter is used when triggering the custom command, sender.source is undefined for both. 

This is test my command now:
Custom command: ```!test-> $_test```
Custom variable: ```$_test-> eval -> execute on isUsed -> change permission to viewers```
```
if (param)
	return "param: "+param+" | source: "+sender.source
else
	return "source: "+sender.source
```

When ```!test``` command is triggered, returns ```source: discord``` & ```source: twitch```
When ```!test stream``` command is triggered, returns ```param: stream | source: undefined``` for both twitch and discord.

I did the script this way instead of just ```return sender.source``` so that a message would actually send to discord. ```return sender.source``` gives error in sogebot console and nothing sent to discord.
